app/parse.py

```diff
@@ -15,8 +48,110 @@ class Product:
     num_of_reviews: int
 
 
+def product_parse(
+        driver: webdriver.Chrome,
+        product_soup: BeautifulSoup
+) -> Product:
+    detailed_url = urljoin(
+        BASE_URL,
+        product_soup.select_one(".title")["href"]
+    )
+    driver.get(detailed_url)
+    title = (
+        driver.find_element(By.CLASS_NAME, "title")
+        .text.strip()
+    )
+    return Product(
+        title=title,
+        description=product_soup.find(
+            "p", {"class": "description"}
+        )
+        .text.strip()
+        .replace("\xa0", " ")
+        .strip(),
+        price=float(
+            product_soup.find("h4", {"class": "price"})
+            .text.strip()
+            .replace("$", "")
+        ),
+        rating=int(len(product_soup.find_all(
+            "span", class_="ws-icon-star"))
+        ),
+        num_of_reviews=int(
+            product_soup.find("p", {"class": "review-count"})
+            .text.strip()
+            .split()[0]
+        ),
+    )
+
+
+def load_full_page(
+        driver: webdriver.Chrome,
+        url: str
+) -> BeautifulSoup:
+    driver.get(url)
+
+    try:
+        accept_button = WebDriverWait(driver, 2).until(
+            exc.element_to_be_clickable((By.CLASS_NAME, "acceptCookies"))
+        )
+        accept_button.click()
+        print("Cookies accepted")
+    except (TimeoutException, NoSuchElementException):
+        print("Cookies not found")
+
+    while True:
+        try:
+            more_button = WebDriverWait(driver, 2).until(
+                exc.presence_of_element_located(
+                    (By.CLASS_NAME, "ecomerce-items-scroll-more")
+                )
+            )
+            if (
+                    "disabled" not in more_button.get_attribute("class")
+                    and more_button.is_displayed()
+            ):
+                more_button.click()
+                time.sleep(2)
+            else:
+                break
+        except (
+                NoSuchElementException,
+                ElementNotInteractableException,
+                TimeoutException,
+        ):
+            break
+
+    return BeautifulSoup(driver.page_source, "html.parser")
+
+
+def get_products(
+        driver: webdriver.Chrome,
+        url: str
+) -> list[Product]:
+    all_products = []
+    soup = load_full_page(driver, url)
+    products = soup.find_all(
+        "div", {"class": "col-md-4 col-xl-4 col-lg-4"}
+    )
+    for product in products:
+        all_products.append(product_parse(driver, product))
+    return all_products
+
+
+def write_to_csv(path_gen: str, products: list) -> None:
+    csv_file = next(path_gen)
```

Comment:
Ensure that the `path_gen` generator is correctly used and that it yields a new path for each call. If the generator is exhausted, it will raise a `StopIteration` exception. Consider handling this exception if there's a possibility of running out of paths.